docs/python/doxygenlib/cdWriterDocstring.py

```diff
@@ -687,49 +699,89 @@ def __getFullDoc(self, pyname, pyobj, doxy):
         # make the doxy element static if it is tagged as such
         if ATTR_STATIC_METHOD in doxy.doc['tags']:
             doxy.static = 'yes'
-        
+
         lines = self.__getShortDescription(pyname, pyobj, doxy)
         if doxy.isFunction() and type(pyobj) != property:
-            lines += self.__getSignatureDescription(pyname, pyobj, doxy)
-            lines.append('')
+            description = self.__getSignatureDescription(pyname, pyobj, doxy)
+            if description is not None:
+                lines += description
+                lines.append('')
         lines += self.__getDocumentation(pyname, pyobj, doxy)
         lines.append('')
         return lines
 
-    def __getOutputFormat(self, pypath, pyobj, overloads):
-        """Return the line that installs the docstring into the namespace."""
+    @classmethod
+    def __stripBoostSig(cls, doc):
+        def looksLikeBoostSig(l):
+            return bool(l and (l[0].isalnum() or l[0] == "_") and ' -> ' in l)
+
+        lines = doc.strip().splitlines()
+        if len(lines) and looksLikeBoostSig(lines[0]):
+            # boost signature has been prepended.  that means if there is
+            # an existing description provided in the C++ wrapper it will be
+            # indented.  strip the signature and dedent the description.
+            found = False
+            newLines = []
+            for line in lines:
+                if not found and line.startswith('    '):
+                    found = True
+                if found:
+                    newLines.append(line)
+            return textwrap.dedent('\n'.join(newLines))
```

Comment:
So, basically we're finding the first indented line, grabbing it and all following, and then dedenting.

I'm assuming you haven't encountered the situation where unindented text follows indented, ie:

```
A
B
    C
    D
E
    F
```
...but we should consider what to do if it happens.  Current behavior would be to return:

```
    C
    D
E
    F
```
...which is likely not desired.

Simplest solution is to just error if we're fairly sure this should never happen, and/or can't make any guesses at what the best behavior would be if it does.  Other options would be to return one of:

```
C
D
F
```
or

```
C
D
E
F
```
...both of which feel more likely to be correct than current behavior.